tensorflow/go/tensor.go

```diff
@@ -94,9 +94,22 @@ func NewTensor(value interface{}) (*Tensor, error) {
 	raw := tensorData(t.c)
 	buf := bytes.NewBuffer(raw[:0:len(raw)])
 	if dataType != String {
-		if err := encodeTensor(buf, val, shape); err != nil {
-			return nil, err
+		if isAllArray(val.Type()) {
```

Comment:
I'm not seeing where scalars are handled in this code. (Am I missing something?)

Reply:
Scalars are handled like a length 1 array. The data pointer in the interface type will point to the memory containing the scalar value just like it would point to the first element of a multi-element array.

Reply:
And more importantly the scalar tests pass!